src/klein/test/test_resource.py

```diff
@@ -810,35 +809,48 @@ def root(request):
 
         @app.handle_errors(TypeError)
         def handle_type_error(request, failure):
-            global type_error_handled
-            type_error_handled = True
-            return
+            type_error_handled[0] = True
 
         @app.handle_errors(TypeFilterTestError)
         def handle_type_filter_test_error(request, failure):
             failures.append(failure)
             request.setResponseCode(501)
-            return
 
         @app.handle_errors
         def handle_generic_error(request, failure):
-            global generic_error_handled
-            generic_error_handled = True
-            return
+            generic_error_handled[0] = True
 
         d = _render(self.kr, request)
 
         self.assertFired(d)
         self.assertEqual(request.processingFailed.called, False)
-        self.assertEqual(type_error_handled, False)
-        self.assertEqual(generic_error_handled, False)
+        self.assertEqual(type_error_handled[0], False)
+        self.assertEqual(generic_error_handled[0], False)
         self.assertEqual(len(failures), 1)
         self.assertEqual(request.code, 501)
 
+        # Verify that above handlers would otherwise have worked
```

Comment:
Maybe add:

```suggestion
        # Verify that above handlers would otherwise have worked.
        # Done to please the code coverage reporting.
```